build.gradle

```diff
@@ -145,9 +145,6 @@ dependencies {
     compileOnly fileTree(dir: knnJarDirectory, include: '*.jar')
     api group: 'org.opensearch', name:'opensearch-ml-client', version: "${opensearch_build}"
     implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.10'
-    // ml-common excluded reflection for runtime so we need to add it by ourselves.
-    // https://github.com/opensearch-project/ml-commons/commit/464bfe34c66d7a729a00dd457f03587ea4e504d9
-    // TODO: Remove following three lines of dependencies if ml-common include them in their jar
     runtimeOnly group: 'org.reflections', name: 'reflections', version: '0.9.12'
     runtimeOnly group: 'org.javassist', name: 'javassist', version: '3.29.2-GA'
     runtimeOnly group: 'org.opensearch', name: 'common-utils', version: "${opensearch_build}"
```

Comment:
why these are not removed if the PR was to remove the direct dependencies.

Reply:
seems that's it's not needed, with recent knn change it works without jar hell error. I'm about to close this PR